site/content/posts/2021-08-31-octant-0.23-release.md

```diff
@@ -0,0 +1,112 @@
+| Succeeded | All containers successfully completed their task and then exited, especially true in the presence of OnFailure [restartPolicy](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#restart-policy) |
+| Failed | All containers terminated and at least one terminated in failure|
+| Unknown| Most likely the node is offline |
+
+there are also three **container states** defined in the Kubernetes API:
+
+| Value| Description|
+| ----| ----|
+|Waiting|   A task is pending to complete, for example a container image download or an init container still running|
+|Running| Container executing with no issues nor pending tasks|
+|Terminated| Container either ran to completion or it failed
+
+With all these possibilities, one could argue that a better approach to diagnose the situation of a Pod is to measure it both at the container and the Pod level.
+
+And that is what the 0.23 version of Octant is enables users to do.
+
+
+# Octant and the Status field
+
+Apart from the **Phase** field that Octant has traditionally displayed for each resource in the Kubernetes environment, the 0.23 release has added the **Status** column which -in the case of Pods- refers specifically to the container State. 
+
+So, if we take a look at the "activator" Pod with Octant:
+
+![](/img/posts/2021/08/18/octantStatus01.png)
+
+We can see that it's in the **Running** phase, but the actual container state is **CrashLoopBackOff** which means that the Kubelet has successfully started it several times but has crashed afterwards without reaching a stable state yet.
+
+Using Octant we can isolate the root cause of the issue:
+
+1. Click on the Pod name
+2. Go to the **Resource Viewer** tab and there you will find the **Last Event** field:
+
+![](/img/posts/2021/08/18/octantStatus02.png)
+
+3. Go to the **Logs** tab to gather even more details around the Readiness Probe failure:
+```json
+   
+{"severity":"ERROR","timestamp":"2021-09-08T20:22:54.283557107Z","logger":"activator","caller":"websocket/connection.go:192","message":"Failed to send ping message to ws://autoscaler.knative-serving.svc.cluster.local:8080","commit":"c75484e","knative.dev/controller":"activator","knative.dev/pod":"activator-67656dcbbb-hq62r","error":"connection has not yet been established"
+```
+
+It looks like the ```activator``` Pod depends on a successful connection to the ```autoscaler``` Pod to start successfully.
+
+4. In the ```autoscaler``` summary page, it can be seen that the Pod requested 1000Mi of memory and, additionally, it has been restarting constantly:
+
+![](/img/posts/2021/08/18/octantStatus04.png)
+
+![](/img/posts/2021/08/18/octantStatus06.png)
+
+5. In this case, the memory request was modified to meet the same value (```100Mi```) all the additional Pods in the app have requested and, in consequence, ```autoscaler``` was able to run and the ```activator``` Pod finally reached the **Running**  state:
```

Comment:
Is this saying setting a request equal to a limit will cause a crash loop or was the image intended to how 100Mi?